### PR TITLE
MINOR; DeleteTopics version tests

### PR DESCRIPTION
**Pull request by jsancio**
Add a DeleteTopics test for all supported versions. Convert the
DeleteTopicsRequestTest to run against both ZK and KRaft mode.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by cmccabe** [comment]
Looks like there is a scala warning (perhaps a difference in scala version behaviors?)
```
[2022-05-09T18:40:33.800Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-12141/core/src/test/scala/unit/kafka/server/DeleteTopicsRequestTest.scala:235:38: the result type of an implicit conversion must be more specific than Object
```

**Review by cmccabe — APPROVED** [review]
LGTM once jenkins is green.

Thanks for tackling this, @jsancio .

File: core/src/test/scala/unit/kafka/server/DeleteTopicsRequestTest.scala [hunk 1]
```diff
@@ -17,24 +17,28 @@
...
 import scala.jdk.CollectionConverters._
 
-class DeleteTopicsRequestTest extends BaseRequestTest {
+class DeleteTopicsRequestTest extends BaseRequestTest with Logging {
 
-  @Test
-  def testValidDeleteTopicRequests(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testValidDeleteTopicRequests(quorum: String): Unit = {
```
Comment (dengziming) [inline core/src/test/scala/unit/kafka/server/DeleteTopicsRequestTest.scala:41]: It seems this is flaky in KRaft mode, we have a `TestUtils.ensureConsistentKRaftMetadata` to fix this kind of flakiness, which waits until the brokers have caught up to the controller metadata topic end offset.

**Review by dengziming — COMMENTED** [review]
Left a suggestion to fix the flakiness

File: core/src/test/scala/unit/kafka/server/DeleteTopicsRequestTest.scala (hunk 1 above)
Comment (jsancio) [reply core/src/test/scala/unit/kafka/server/DeleteTopicsRequestTest.scala:41]: Thank @dengziming . I added the check. Let's see if the tests pass on Jenkins.

File: core/src/test/scala/unit/kafka/server/DeleteTopicsRequestTest.scala [hunk 2]
```diff
@@ -66,6 +70,11 @@ class DeleteTopicsRequestTest extends BaseRequestTest {
     val response = sendDeleteTopicsRequest(request)
     val error = response.errorCounts.asScala.find(_._1 != Errors.NONE)
     assertTrue(error.isEmpty, s"There should be no errors, found ${response.data.responses.asScala}")
+
+    if (isKRaftTest()) {
```
Comment (dengziming) [inline core/src/test/scala/unit/kafka/server/DeleteTopicsRequestTest.scala:74]: Tests all passed after this change.

**Review by dengziming — APPROVED** [review]
LGTM.

**Pull request merged**